### PR TITLE
Completed and verified model test

### PR DESCRIPTION
**Pull request by GBowman1**
## Describe Changes Made
In this pr I added test to verify query methods in the merchant model, as of now test pass and everything seems correct I would appreciate a second set of eyes to do a code review.

We should add test to dashboard index view to verify correct data in table data as well, but in theory if the model test are correct then so should the table data.

I added factory bots for the other models in the spec/factories directory

## If PR is linked to a Issue, Add Issue
Completely Finished US 3 & 4

## PR Checklist
- [x] Added Reviewer
- [x] Followed TDD, And Test are Passing
- [x] Linked Issues for Project Board

### CODE REVIEW

File: spec/factories/invoice_item_factory.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FactoryBot.define do
+    factory :invoice_item do
+        item
+        invoice
+        quantity { rand(1..10) }
+        unit_price { Faker::Number.number(digits: 4)}
```
Comment (JCL461437) [inline spec/factories/invoice_item_factory.rb:6]: Nice use of the Faker

File: spec/factories/item_factory.rb [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
-#         description { Faker::Lorem.sentence }
-#         unit_price { Faker::Number.decimal(l_digits: 2) }
-#         merchant
-#     end
-# end
\ No newline at end of file
+FactoryBot.define do
+    factory :item do
+        name { Faker::Commerce.product_name }
+        description { Faker::Lorem.sentence }
```
Comment (JCL461437) [inline spec/factories/item_factory.rb:4]: Again I like the Faker usage

File: spec/models/merchant_spec.rb [hunk 3]
```diff
@@ -8,51 +8,97 @@
...
-        invoice_item_3 = InvoiceItem.create!(item_id: item_3.id, invoice_id: invoice_3.id, quantity: 1, unit_price: 3000, status: 2)
-        invoice_item_4 = InvoiceItem.create!(item_id: item_4.id, invoice_id: invoice_4.id, quantity: 1, unit_price: 4000, status: 0)
-        invoice_item_5 = InvoiceItem.create!(item_id: item_5.id, invoice_id: invoice_5.id, quantity: 1, unit_price: 5000, status: 0)
-
-        transaction_1 = Transaction.create!(invoice_id: invoice_1.id, credit_card_number: 1234567890123456, result: 0)
-        transaction_2 = Transaction.create!(invoice_id: invoice_2.id, credit_card_number: 1234567890123456, result: 0)
-        transaction_3 = Transaction.create!(invoice_id: invoice_3.id, credit_card_number: 1234567890123456, result: 0)
-        transaction_4 = Transaction.create!(invoice_id: invoice_4.id, credit_card_number: 1234567890123456, result: 0)
-        transaction_5 = Transaction.create!(invoice_id: invoice_5.id, credit_card_number: 1234567890123456, result: 0)
+      let(:merchant) { create(:merchant) }
```
Comment (JCL461437) [inline spec/models/merchant_spec.rb:11]: These tests look great to me Garrett! That said I have a much more imperfect understanding than you but I would say this looks good.

File: spec/models/merchant_spec.rb [hunk 4]
```diff
@@ -8,51 +8,97 @@
...
+          invoice_2 = create(:invoice, customer: customer_2, items: [item_1])
+          invoice_3 = create(:invoice, customer: customer_3, items: [item_1])
+          invoice_4 = create(:invoice, customer: customer_4, items: [item_1])
+          invoice_5 = create(:invoice, customer: customer_5, items: [item_1])
+          invoice_6 = create(:invoice, customer: customer_6, items: [item_1])
+          invoice_7 = create(:invoice, customer: customer_1, items: [item_1])
+          invoice_8 = create(:invoice, customer: customer_1, items: [item_1])
+          invoice_9 = create(:invoice, customer: customer_2, items: [item_1])
+
+          create(:transaction, invoice: invoice_1, result: 0)
```
Comment (JCL461437) [inline spec/models/merchant_spec.rb:52]: I like the dynamic nature of this handrolled data. Great work

File: spec/models/merchant_spec.rb [hunk 5]
```diff
@@ -8,51 +8,97 @@
...
+          create(:transaction, invoice: invoice_2, result: 0)
+          create(:transaction, invoice: invoice_3, result: 0)
+          create(:transaction, invoice: invoice_4, result: 0)
+          create(:transaction, invoice: invoice_5, result: 0)
+          create(:transaction, invoice: invoice_6, result: 1)
+          create(:transaction, invoice: invoice_7, result: 0)
+          create(:transaction, invoice: invoice_8, result: 1)
+          create(:transaction, invoice: invoice_9, result: 1)
+
```
Comment (JCL461437) [inline spec/models/merchant_spec.rb:97]: I bet it was tedious, but I certainly worth the effort to have such control over your data

**Review by JCL461437 — APPROVED** [review]

File: spec/models/merchant_spec.rb (hunk 3 above)
Comment (austincarrjones) [reply spec/models/merchant_spec.rb:11]: Yeah cool to see how the factorybot can be used to refractor long test setup

**Review by austincarrjones — APPROVED** [review]

**Pull request merged**